examples/cifar10_resnet.py

```diff
@@ -69,87 +100,271 @@
+    x = Activation('relu')(x)
+    x = Conv2D(num_filters,
+               kernel_size=kernel_size,
+               strides=strides,
+               padding='same',
+               kernel_initializer='he_normal',
+               kernel_regularizer=l2(1e-4))(x)
+    return x
+
+
+def resnet_v1(input_shape, depth, num_classes=10):
+    """ResNet Version 1 Model builder [a]
+
+    Stacks of 2 x (3 x 3) Conv2D-BN-ReLU
+    Last ReLU is after the shortcut connection.
+    The number of filters doubles when the feature maps size
+    is halved.
+    The Number of parameters is approx the same as Table 6 of [a]:
+    ResNet20 0.27M
+    ResNet32 0.46M
+    ResNet44 0.66M
+    ResNet56 0.85M
+    ResNet110 1.7M
+
+    # Arguments
+        input_shape (tensor): shape of input image tensor
+        depth (int): number of core convolutional layers
+        num_classes (int): number of classes (CIFAR10 has 10)
+
+    # Returns
+        model (Model): Keras model instance
+    """
+    assert (depth - 2) % 6 == 0, 'depth should be 6n+2 (eg 20, 32, 44 in [a])'
+    # Start model definition.
+    inputs = Input(shape=input_shape)
+    num_filters = 16
+    num_sub_blocks = int((depth - 2) / 6)
+
+    x = conv_bn_relu(inputs=inputs)
+    # Instantiate convolutional base (stack of blocks).
+    for i in range(3):
+        for j in range(num_sub_blocks):
+            strides = 1
+            is_first_layer_but_not_first_block = j == 0 and i > 0
+            if is_first_layer_but_not_first_block:
+                strides = 2
+            y = conv_bn_relu(inputs=x,
+                             num_filters=num_filters,
+                             strides=strides)
+            y = conv_bn(inputs=y,
+                        num_filters=num_filters)
+            if is_first_layer_but_not_first_block:
+                x = conv_bn(inputs=x,
+                            num_filters=num_filters,
+                            kernel_size=1,
+                            strides=strides)
+            x = keras.layers.add([x, y])
+            x = Activation('relu')(x)
+        num_filters = 2 * num_filters
+
+    # Add classifier on top.
+    # v1 does not use BN after last shortcut connection-ReLU
+    x = AveragePooling2D(pool_size=8)(x)
+    y = Flatten()(x)
+    outputs = Dense(num_classes,
+                    activation='softmax',
+                    kernel_initializer='he_normal')(y)
+
+    # Instantiate model.
+    model = Model(inputs=inputs, outputs=outputs)
+    return model
+
+
+def resnet_v2(input_shape, depth, num_classes=10):
+    """ResNet Version 2 Model builder [b]
+
+    Stacks of (1 x 1)-(3 x 3)-(1 x 1) BN-ReLU-Conv2D or also known as
+    bottleneck layer
+    First shortcut connection per layer is 1 x 1 Conv2D.
+    Second and onwards shortcut connection is identity.
+    Features maps sizes: 16(input), 64(1st sub_block), 128(2nd), 256(3rd)
+    Torch Reference Code:
+    https://github.com/KaimingHe/resnet-1k-layers/blob/master/resnet-pre-act.lua
```

Comment:
This is unlicensed code, please don't include a reference to it. Reference to the paper should be enough.

Reply:
Removed. Thanks.